### DISCUSSION TITLE
Converting many images to dds filetype

### DISCUSSION THREAD

**Discussion by Zeno-Heartfell** [ImageMagick/ImageMagick#5045] [Help] [answered]
This is my first time asking anything like this so sorry if I do anything wrong. I am trying to convert a bunch of texture files, both .tga and .bmp, of Elder Scroll 3 Morrowind to dds so I can run OpenMW correctly.

The code I cobbled together after 3 days of searching is such:

magick mogrify -path "D:\path\to\files" -format dds * .tga

When run I get dozons of lines reading

mogrify: no decode delegate for this image format 'DLL' @ error/constitute.c/ReadImage/737.

mogrify: unable to open moduale file 'C:\Program Files\ImageMagick-7.1.0-Q16-HDRI\modules\coders\IM_MOD_RL_DLL_.dll':ng/module.c/GetMagickModulePath/675

I installed ImageMagick with all optional addons available. I don't know where to go from here. Please help me, thank you!

**Comment by snibgo** [comment]
```
magick mogrify -path "D:\path\to\files" -format dds * .tga
```
You have a space between "\*" and ".tga". Why? The program will first look for files matching "\*" in your current directory, and I suppose that has *.dll files. IM tells you these are not image files. Then it will look for files matching exactly ".tga", and there probably are none.

As you probably know, "-path" is used for the place to store output files, not to read input files.

**Reply by Zeno-Heartfell** [reply]
I tried putting the space since whenever I didn't it came back that it couldn't find the file. No I did not know that -path was for output file location, but I do need them to go there anyway. How can I make the program look at a specific location only. I dont want all of the files on my PC changed to dds. This is the first time ever using ImageMagick.

**Comment by snibgo** [comment, marked as answer]
You have no files that match *.tga in your current directory. You can either change your current directory using the "cd" or "pushd" command, or specify the input directory in your mogrify command.

For help on "cd", type "help cd" at the command line.

To specify the input directory:
```
magick mogrify -path "D:\path\to\outputfiles" -format dds D:\path\to\inputfiles\*.tga
```
The two directories can be the same. If they are the same, "-path" is redundant.

**Reply by Zeno-Heartfell** [reply]
Thank you so much for your help. I managed to convert the files with this code line.

D:\>Magick mogrify -format dds "D:\path\to\files\*.that

However for some reason it wouldn't work without the single quote. Why is that?

**Comment by snibgo** [comment]
If you have open-quote, you should also have a close-quote.

I suspect you have spaces in your filenames. The shell uses spaces to separate command-line arguments. So when an argument (such as a filename) contains a space, the quote tells the shell not to consider the space as a separator.

**Reply by Zeno-Heartfell** [reply]
I do have spaces in the file names. I tried with a close quote in various places but it wouldn't work. I managed to convert all the files I needed to so thank you very much for all your help. I now have another issue with upside-down texturea but I think I'll work on that myself before I open another help thread.
